gymnasium/vector/sync_vector_env.py

```diff
@@ -243,7 +243,21 @@ def step(
 
         Returns:
             The batched environment step results
+
+        Raises:
+            ValueError: If the number of actions does not match the number of environments.
         """
+        try:
```

Comment:
As we support non array actions, e.g., a dictionary of actions then we can't use this code. 

We should implement the same code as `async_vector_env`, then `for i, (action, _) in enumerate(zip(actions, self.envs, strict=True)):`

Then could you add a test in `tests/vector/test_vector_env.py` with both `SyncVectorEnv` and `AsyncVectorEnv`